src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts

```diff
@@ -0,0 +1,164 @@
+        this.lectureUnit = undefined;
+        this.exercise = undefined;
+    }
+
+    onPrevTask() {
+        // reset state to avoid invalid states
+        this.undefineAll();
+        if (this.learningPathHistoryStorageService.hasPrevious(this.learningPathId)) {
+            const entry = this.learningPathHistoryStorageService.getPrevious(this.learningPathId);
+            if (entry instanceof LectureUnitEntry) {
+                this.learningObjectId = entry.lectureUnitId;
+                this.lectureId = entry.lectureId;
+                this.loadLectureUnit();
+            } else if (entry instanceof ExerciseEntry) {
+                this.learningObjectId = entry.exerciseId;
+                this.loadExercise();
+            }
+        }
+    }
+
+    loadLectureUnit() {
+        this.lectureService.findWithDetails(this.lectureId!).subscribe({
+            next: (findLectureResult) => {
+                this.lecture = findLectureResult.body!;
+                if (this.lecture?.lectureUnits) {
+                    this.lectureUnit = this.lecture.lectureUnits.find((lectureUnit) => lectureUnit.id === this.learningObjectId);
+                }
+            },
+            error: (errorResponse: HttpErrorResponse) => onError(this.alertService, errorResponse),
+        });
+        this.router.navigate(['lecture-unit'], { relativeTo: this.activatedRoute });
+    }
+
+    loadExercise() {
+        this.exerciseService.getExerciseDetails(this.learningObjectId).subscribe({
+            next: (exerciseResponse) => {
+                this.exercise = exerciseResponse.body!;
+            },
+            error: (errorResponse: HttpErrorResponse) => onError(this.alertService, errorResponse),
+        });
+        this.router.navigate(['exercise'], { relativeTo: this.activatedRoute });
+    }
+
+    /**
+     * This function gets called if the router outlet gets activated. This is
+     * used only for the LearningPathLectureUnitViewComponent
+     * @param instance The component instance
+     */
+    onChildActivate(instance: LearningPathLectureUnitViewComponent | CourseExerciseDetailsComponent) {
+        if (instance instanceof LearningPathLectureUnitViewComponent) {
+            this.setupLectureUnitView(instance);
+        } else {
+            this.setupExerciseView(instance);
+        }
+    }
+
+    setupLectureUnitView(instance: LearningPathLectureUnitViewComponent) {
+        if (this.lecture) {
+            instance.lecture = this.lecture;
+            instance.lectureUnit = this.lectureUnit!;
+        }
+    }
+
+    setupExerciseView(instance: CourseExerciseDetailsComponent) {
+        if (this.exercise) {
+            instance.learningPathMode = true;
+            instance.courseId = this.courseId;
+            instance.exerciseId = this.learningObjectId;
+        }
+    }
+
+    onNodeClicked(node: NgxLearningPathNode) {
+        if (node.type === NodeType.LECTURE_UNIT || node.type === NodeType.EXERCISE) {
```

Comment:
Let's return early to avoid nesting the entire function

```suggestion
        if (node.type !== NodeType.LECTURE_UNIT && node.type !== NodeType.EXERCISE) {
            return;
        }
```